### PR TITLE
Validate amp-app-banner url in ios

### LINKED ISSUE #9180: Validate amp-app-banner url in ios
Validate the `app-argument` inside `<meta name="apple-itunes-app" ... >`

### PR DESCRIPTION
**Pull request by muxin**
Fix #9180

### CODE REVIEW

File: src/url.js [hunk 1]
```diff
@@ -321,6 +321,19 @@ export function isLocalhostOrigin(url) {
...
 /**
+ * Returns whether the URL has valid protocol.
+ * @param {string|!Location} url URL of an AMP document.
+ * @return {boolean}
+ */
+export function isProtocolValid(url) {
+  if (typeof url == 'string') {
+    url = parseUrl(url);
+  }
+  const invalidProtocols = ['javascript:', 'data:', 'vbscript:'];
```
Comment (aghassemi) [inline src/url.js]: move outside of function as  `const INVALID_PROTOCOLS = `

File: src/url.js [hunk 2]
```diff
@@ -321,6 +321,19 @@ export function isLocalhostOrigin(url) {
 }
 
 /**
+ * Returns whether the URL has valid protocol.
+ * @param {string|!Location} url URL of an AMP document.
+ * @return {boolean}
+ */
+export function isProtocolValid(url) {
```
Comment (aghassemi) [inline src/url.js]: please add bunch of tests for this in a separate PR today (let's merge this PR to cherry pick)

File: extensions/amp-app-banner/0.1/amp-app-banner.js [hunk 3]
```diff
@@ -283,6 +283,10 @@ export class AmpIosAppBanner extends AbstractAppBanner {
 
     const appId = config['app-id'];
     const openUrl = config['app-argument'];
+
+    user().assert(isProtocolValid(openUrl),
+        'The url in app-argument is invalid');
```
Comment (aghassemi) [inline extensions/amp-app-banner/0.1/amp-app-banner.js]:
```
user().assert(isProtocolValid(openUrl), ' %s: The url in app-argument is invalid', TAG);
```

**Comment by aghassemi** [comment]
@muxin both url.js and test file need `/*eslint no-script-url: 0*/` to pass lint

File: extensions/amp-app-banner/0.1/amp-app-banner.js (hunk 3 above)
Comment (muxin) [inline extensions/amp-app-banner/0.1/amp-app-banner.js]: done.

File: src/url.js [hunk 4]
```diff
@@ -321,6 +321,22 @@ export function isLocalhostOrigin(url) {
...
+  if (typeof url == 'string') {
+    url = parseUrl(url);
+  }
+  const invalidProtocols = [
```
Comment (jridgewell) [inline src/url.js]: Hoist this array.

**Comment by aghassemi** [comment]
@muxin merging this, please send a PR for the remaining review comments, thanks.

**Pull request merged**